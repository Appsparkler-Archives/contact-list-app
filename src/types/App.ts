import React from "react";
import { IContactFormData } from "./data";

export interface IAppProps {
  contacts: IContactFormData[];
  onCreate: (newContact: IContactFormData) => void;
  onEdit: (editedContact: IContactFormData) => void;
  onDelete: (contactIdToDelete: IContactFormData["id"]) => void;
}

export type TAppFC = React.FC<IAppProps>;
