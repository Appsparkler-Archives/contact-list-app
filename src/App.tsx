import { Box, TextField, Typography } from "@mui/material";
import { TAppFC } from "types";
import { FiltersChip } from "components/FilterChip/FiltersChip";
import { ContactListAccordion } from "components/ContactListAccordion/ContactListAccordion";
import { CreateContactForm } from "components/CreateContactForm/CreateContactForm";

export const App: TAppFC = ({ contacts, onCreate, onEdit, onDelete }) => {
  return (
    <Box
      display={"flex"}
      flexDirection={"column"}
      p={2}
      height={"100vh"}
      sx={{ backgroundColor: "grey.50", position: "relative" }}
      gap={2}
    >
      <Typography variant="h4">Contacts</Typography>
      <TextField
        placeholder="Search..."
        sx={{ backgroundColor: "background.paper" }}
      />
      {contacts.length > 0 && <FiltersChip contacts={contacts} />}
      <Box>
        <CreateContactForm onCreate={onCreate} />
      </Box>
      <ContactListAccordion
        contacts={contacts}
        onEdit={onEdit}
        onDelete={onDelete}
      />
    </Box>
  );
};

export default App;
