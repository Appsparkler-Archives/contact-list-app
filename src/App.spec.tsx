import { render } from "@testing-library/react";
import App from "App";

const onCreate = jest.fn();
const onEdit = jest.fn();
const onDelete = jest.fn();

const AppRenderer = () => (
  <App contacts={[]} onCreate={onCreate} onEdit={onEdit} onDelete={onDelete} />
);

describe("App - Snapshot Tests", () => {
  it("should render App component", () => {
    const { container } = render(<AppRenderer />);
    expect(container).toMatchSnapshot();
  });
});
