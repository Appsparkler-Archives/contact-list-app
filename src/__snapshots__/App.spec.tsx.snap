// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - Snapshot Tests should render App component 1`] = `
<div>
  <div
    class="MuiBox-root css-ejw0ia"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Contacts
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
    >
      You have 
      0
       contacts.
    </h6>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonAddAlt1Icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 8c0-2.21-1.79-4-4-4S5 5.79 5 8s1.79 4 4 4 4-1.79 4-4zm2 2v2h3v3h2v-3h3v-2h-3V7h-2v3h-3zM1 18v2h16v-2c0-2.66-5.33-4-8-4s-8 1.34-8 4z"
            />
          </svg>
        </span>
        Add New Contact
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div />
  </div>
</div>
`;
